src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -326,5 +335,21 @@ def getRequestSummaryWithJobInfo(self, requestName):
         self._updateRequestInfoWithJobInfo(requestInfo)
         return requestInfo
         
+    def getArchivedRequests(self):
+        """
+        get list of archived workflow in wmstats db.
+        """
+        
+        options = {"group_level": 1, "reduce": True}
+        
+        results = self.couchDB.loadView(self.couchapp, "allWorkflows", options = options)['rows']
+        requestNames = [x['key'] for x in results]
+        
+        workflowDict = self.reqDB.getStatusAndTypeByRequest(requestNames)
```

Comment:
It shouldn't be that many, 
1. it use reduce so only workflows it will return the number of workflows not number of docs.
2. and once the document related workflow is deleted it is gone from db, so next polling cycle there will be less documents to retrieve.

i.e. initially there are 2000 archived workflows (maybe 2 million docs) and 1000 active workflows (with million doc).
after deleting the will be only 1000 active workflows but 2 million docs will be deleted.
Next cycle there might be few more workflows  are archived. (I run this twice a day, it shouldn't be creating too many archived workflows in have a day)
https://github.com/dmwm/deployment/blob/master/reqmon/config.py#L106